runtime/doc/lua-guide.txt

```diff
@@ -0,0 +1,714 @@
+functions have to be called with this syntax: >lua
+
+    vim.fn['my#autoload#function']()
+
+------------------------------------------------------------------------------
+See also
+
+• |builtin-functions| - alphabetic list of all Vim script functions
+• |function-list| - list of all Vim script functions grouped by topic
+
+==============================================================================
+Setting variables
+
+The most convenient way to set vim variables is to use the `vim.*` wrappers:
+
+• |vim.g|   - global variables (|g:|)
+• |vim.b|   - buffer variables (|b:|)
+• |vim.w|   - window variables (|w:|)
+• |vim.t|   - tabpage variables (|t:|)
+• |vim.v|   - predefined Vim variables (|v:|)
+• |vim.env| - environment variables >lua
+
+    vim.g.some_global_variable = {
+        key1 = "value",
+        key2 = 300
+    }
+
+    print(vim.inspect(vim.g.some_global_variable)) -- { key1 = "value", key2 = 300 }
+
+    -- target a specific buffer/window/tabpage
+    vim.b[2].myvar = 1
+
+Some variable names may contain characters that cannot be used for identifiers
+in Lua. You can still manipulate these variables by using this syntax:
+`vim.g['my#variable']`.
+
+To delete one of these variables, simply assign `nil` to it: >lua
+
+    vim.g.some_global_variable = nil
+
+See also:
+- |lua-vim-variables|
+
+==============================================================================
+Setting options
+
+The simplest way to set options is to use the `vim.*o` and `vim.opt*` Lua API
+wrappers. These are wrappers for the API functions described in |api-options|.
+
+------------------------------------------------------------------------------
+vim.*o
+
+These wrappers allow you to manipulate options as if they were variables:
+
+• |vim.o|  - behaves like |:set|
+• |vim.go| - behaves like |:setglobal|
+• |vim.bo| - behaves like |:setlocal| for buffer-local options
+• |vim.wo| - behaves like |:setlocal| for window-local options >lua
+
+    vim.o.smarttab = false -- :set nosmarttab
+    print(vim.o.smarttab) -- false
+    vim.o.isfname = vim.o.isfname .. ',@-@' -- on Linux: :set isfname+=@-@
+    print(vim.o.isfname) -- '@,48-57,/,.,-,_,+,,,#,$,%,~,=,@-@'
+
+    vim.bo.shiftwidth = 4
+    print(vim.bo.shiftwidth) -- 4
+
+You can specify a buffer or window number for buffer-local and window-local
+options. If no number is given, the current buffer/window is used: >lua
+
+    vim.bo[4].expandtab = true -- sets expandtab to true in buffer 4
+    vim.wo.number = true -- sets number to true in current window
+<
+
+------------------------------------------------------------------------------
+vim.opt*
+
+These wrappers are a more programmatic way to set options which are more
+suited for plugin authors.
+
+TODO: can use tags after https://github.com/neovim/neovim/pull/21138 is
+merged.
+
+• |vim.opt|: behaves like |:set|
+• vim.opt_global: behaves like |:setglobal|
+• vim.opt_local: behaves like |:setlocal| >lua
```

Comment:
```suggestion
• |vim.opt|: behaves like |:set|
• |vim.opt_global|: behaves like |:setglobal|
• |vim.opt_local|: behaves like |:setlocal| >lua
```
I just merged it :)

(And `opt_local` _does_ behave like `:setlocal` also for window-local options. Maybe it would make sense here to start this section with `vim.opt` and then introduce `vim.*o` as an advanced, more direct, "variable-like" access?